cmd/meroxa/root/apps/deploy.go

```diff
@@ -105,23 +112,65 @@ func (d *Deploy) getDockerHubAccessTokenEnv() string {
 	return d.config.GetString(dockerHubAccessTokenEnv)
 }
 
-func (d *Deploy) checkRequiredEnvVars() error {
-	if d.getDockerHubUserNameEnv() == "" || d.getDockerHubAccessTokenEnv() == "" {
-		msg := fmt.Sprintf("both %q and %q are required to be set to deploy your application", dockerHubUserNameEnv, dockerHubAccessTokenEnv)
-		return errors.New(msg)
+func (d *Deploy) validateDockerHubFlags() error {
+	if d.flags.DockerHubUserName != "" {
+		d.goDeploy.DockerHubUserNameEnv = d.flags.DockerHubUserName
+		if d.flags.DockerHubAccessToken == "" {
+			return errors.New("--docker-hub-access-token is required when --docker-hub-username is present")
+		}
 	}
 
-	d.goDeploy.DockerHubUserNameEnv = d.getDockerHubUserNameEnv()
-	d.goDeploy.DockerHubAccessTokenEnv = d.getDockerHubAccessTokenEnv()
+	if d.flags.DockerHubAccessToken != "" {
+		d.goDeploy.DockerHubAccessTokenEnv = d.flags.DockerHubAccessToken
+		if d.flags.DockerHubUserName == "" {
+			return errors.New("--docker-hub-username is required when --docker-hub-access-token is present")
+		}
+	}
+	return nil
+}
 
+func (d *Deploy) validateDockerHubEnvVars() error {
+	if d.getDockerHubUserNameEnv() != "" {
+		d.goDeploy.DockerHubUserNameEnv = d.getDockerHubUserNameEnv()
+		if d.getDockerHubAccessTokenEnv() == "" {
+			return fmt.Errorf("%s is required when %s is present", dockerHubAccessTokenEnv, dockerHubUserNameEnv)
+		}
+	}
+	if d.getDockerHubAccessTokenEnv() != "" {
+		d.goDeploy.DockerHubAccessTokenEnv = d.getDockerHubAccessTokenEnv()
+		if d.getDockerHubUserNameEnv() == "" {
+			return fmt.Errorf("%s is required when %s is present", dockerHubUserNameEnv, dockerHubAccessTokenEnv)
+		}
+	}
+	return nil
+}
+
+func (d *Deploy) validateLocalDeploymentConfig() error {
+	// Check if users had an old configuration where DockerHub credentials were set as environment variables
+	err := d.validateDockerHubEnvVars()
+	if err != nil {
+		return err
```

Comment:
Could we also log out the error here using `logger.Error` onto the console or is this already taken care of by returning the `err` variable?

Reply:
@jayjayjpg It's been taken care of. That function would return the error to then to the Execute function which Cobra will print out as an error.

Reply:
Got it and thank you for the outline, that's super helpful!